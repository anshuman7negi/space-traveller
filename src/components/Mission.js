import PropTypes from 'prop-types';
import classNames from 'classnames';
import MissionButton from './MissionButton';

const Mission = ({
  id, missionName, description, highlights, reserved,
}) => {
  const rows = classNames('border pl-2', {
    'bg-gray-100': highlights,
  });
  return (
    <>
      <tr className={rows}>
        <td className="border pl-5 pt-5 align-top">{ missionName }</td>
        <td className="border p-5 text-justify">{ description }</td>
        <td className="border pl-5">
          {reserved ? (
            <span className="inline-flex items-center px-2.5 py-0.5 rounded text-xs font-medium bg-green-500 text-white">
              Active Member
            </span>
          ) : (
            <span className="inline-flex items-center px-2.5 py-0.5 rounded text-xs font-medium bg-gray-500 text-white">
              NOT A MEMBER
            </span>
          )}
        </td>
        <td className="border pl-5" aria-hidden="true">
          <MissionButton id={id} reserved={reserved} />
        </td>
      </tr>
    </>
  );
};

export default Mission;

Mission.propTypes = {
  id: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  highlights: PropTypes.bool.isRequired,
  reserved: PropTypes.bool.isRequired,
};
