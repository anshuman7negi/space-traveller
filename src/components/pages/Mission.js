const Mission = () => {
  <>
    <tr>
      <td> Mission Name </td>
      <td> Description </td>
      <td>
        <span>Active Member</span>
        <span>Not A Member</span>
      </td>
      <td>
        <span>Mission Button</span>
      </td>
    </tr>
  </>;
};

Mission();

export default Mission;
