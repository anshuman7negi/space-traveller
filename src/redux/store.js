import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
