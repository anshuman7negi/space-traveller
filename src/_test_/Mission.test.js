import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Mission from '../components/Mission';

test('Mission component renders correctly', () => {
  const missionProps = {
    id: 'mission-123',
    missionName: 'Sample Mission',
    description: 'A sample mission description.',
    highlights: true,
    reserved: false,
  };

  const tree = renderer.create(
    <Provider store={store}>
      <Mission
        id={missionProps.id}
        missionName={missionProps.missionName}
        description={missionProps.description}
        highlights={missionProps.highlights}
        reserved={missionProps.reserved}
      />
    </Provider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
