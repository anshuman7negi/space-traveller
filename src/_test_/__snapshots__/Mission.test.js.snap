// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component renders correctly 1`] = `
<tr
  className="border pl-2 bg-gray-100"
>
  <td
    className="border pl-5 pt-5 align-top"
  >
    Sample Mission
  </td>
  <td
    className="border p-5 text-justify"
  >
    A sample mission description.
  </td>
  <td
    className="border pl-5"
  >
    <span
      className="inline-flex items-center px-2.5 py-0.5 rounded text-xs font-medium bg-gray-500 text-white"
    >
      NOT A MEMBER
    </span>
  </td>
  <td
    aria-hidden="true"
    className="border pl-5"
  >
    <button
      className="text-gray-500 bg-transparent cursor-pointer mt-3 rounded px-[1.2rem] py-2 font-medium outline outline-gray-400"
      onClick={[Function]}
      type="button"
    >
      Join missions
    </button>
  </td>
</tr>
`;
